### DISCUSSION TITLE
Variables - Evaluate regardless of the field they are in or datasource used

### DISCUSSION THREAD

**Discussion by Rodan07** [grafana/grafana#45741] [General]
The problem is that I would like to be able to use values that are stored in variable to change graph heights or set thresholds. In my particular case, based on the bandwidth speed available at a site. That way I can see at a glance if some site is using up there available bandwidth and if I should either upgrade the service if available or look for odd usage at a site.

It would also be nice if I could also use the query to do the same thing something like ${A.maxBandwidth} for a Axis and then a math funtion like 80% threshold is (${A.maxBandwidth}*.8). 

I have been using the infinity plugin a lot as I get most of my data via OpenAPIs that return JSON and that seems to be the best plugin for that use case. I have noticed that not all variables are converted to their proper data in those cases and you can't use a previous query as a variable at all. 

It would be nice if the conversions of variables and query refids were done regardless of what the datasource or form field was. I guess I am used to programing where the language does the conversion of variables and it is not up to individual modules or plugins. 

Thank you for all you guys do it is a great program.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
